lms/views/admin/application_instance.py

```diff
@@ -264,12 +271,33 @@ def show_instance(self):
     def update_instance(self):
         ai = self._get_ai_or_404(**self.request.matchdict)
 
+        # Keep the same org if no public_id is sent
+        organization_id = ai.organization_id
+        if org_public_id := self.request.params.get("org_public_id", "").strip():
+            try:
+                if org := self.organization_service.get_by_public_id(org_public_id):
+                    organization_id = org.id
+                else:
+                    raise ValidationError(
+                        messages={
+                            "organization_id": [
+                                f"Organization {org_public_id} not found"
+                            ]
+                        }
+                    )
+            except ValidationError as err:
+                self.request.session.flash(err.messages, "validation")
+                return HTTPFound(
+                    location=self.request.route_url("admin.instance.id", id_=ai.id)
+                )
+
         self.application_instance_service.update_application_instance(
             ai,
             lms_url=self.request.params.get("lms_url", "").strip(),
             deployment_id=self.request.params.get("deployment_id", "").strip(),
             developer_key=self.request.params.get("developer_key", "").strip(),
             developer_secret=self.request.params.get("developer_secret", "").strip(),
+            organization_id=organization_id,
```

Comment:
It could be argued that the AI service itself could take the public_id and deal with finding it itself.

I reckon for the admin pages this makes sense. If we see this being done more often we could look into refactoring it into the service.